### PR TITLE
Don't explicitly link to libstdc++

### PR DESCRIPTION
**Pull request by isuruf**
Let the compiler do it for you.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,8 +94,6 @@ if(CMAKE_COMPILER_IS_GNUCXX OR (CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
       if(BUILD_WITH_CLANG_LIBRARIES)
          set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -nostdinc++ -I${BUILD_WITH_CLANG_LIBRARIES}/include/c++/v1 -DCLANG_LIBRARIES")
          set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -Wl,-rpath,${BUILD_WITH_CLANG_LIBRARIES}/lib -L${BUILD_WITH_CLANG_LIBRARIES}/lib")
-      else()
-         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -lstdc++")
```
Comment (DavidTruby) [inline CMakeLists.txt]: If we want to preserve the behaviour of actually choosing libstdc++ here over libc++ then this should say -stdlib=libstdc++. Just leaving off the flag would change the behaviour on macOS and with some builds of clang I believe. We definitely shouldn't link to it with `-l` though, good spot!

Comment (isuruf) [reply CMakeLists.txt]: Shouldn't that be left to the user though? On MacOS, the user should specify `MACOS_DEPLOYMENT_TARGET` to select the correct C++ library. On Windows, the build type determines the C++ library and it is neither libstdc++ nor libc++.

Comment (DavidTruby) [reply CMakeLists.txt]: Possibly. I'm not sure we should really be playing with this at all (as in, all this BUILD_WITH_CLANG_LIBRARIES, -gcc-toolchain etc logic), the user can select a specific setup with -DCMAKE_CXX_FLAGS if they really want to

Comment (DavidTruby) [reply CMakeLists.txt]: I think this will all go away when #969 is merged anyway, but @foxpm will know more about that

Comment (pmccormick) [reply CMakeLists.txt]: @isuruf, good catch.   I'll look at this a bit more closely before I issue a new PR on some updated cmake configuration (hopefully later today).  Thanks.

Comment (DavidTruby) [reply CMakeLists.txt]: I don't think we should need our own configuration on any of this (standard library selection/compiler selection) as it's all handled in the top level llvm CMakeLists.txt, which should trickle down to us once we're using that right?

Comment (pmccormick) [reply CMakeLists.txt]: Yes, I believe you are correct.  My new PR should have this all removed (testing it now).

**Comment by DavidTruby** [comment]
This should be resolved by #1045 could you check if you see the same problem with that PR? Thanks!

**Comment by isuruf** [comment]
Closing in favour of #1045

**Pull request closed without merging**